src/Agent.Worker/TaskRunner.cs

```diff
@@ -673,5 +681,29 @@ private void PublishTelemetry(Definition taskDefinition, HandlerData handlerData
                 ExecutionContext.Debug(ex.ToString());
             }
         }
+
+        private bool IsTaskSDKTokenRequired(IHandler handler, Definition task)
```

Comment:
This is pretty complex, some diagnostics tracing might be helpful if something goes wrong

Reply:
Done